### ISSUE TITLE
How can I know the maximum zoom level?

### ISSUE THREAD

**Issue by Ricardo1980** [Kjuly/MKMapView-ZoomLevel#1]
Hello!
What I what is that everytime the user taps a pin, this pin is centered and zoomed one unit, but when zoom is maximum, I want to show a callout or jump to another view.
So, in iorder to implement this logic, how can I know the maximum zoom level?
Thanks.

**Comment by Kjuly** [comment]
Hi @Ricardo1980 , generally, the max zoomLevel is 20 (from 0 to 20). But seems it can only be reached to 17. You can check it yourself, just print the `MKMapView+ZoomLevel`'s `-zoomLevel` every time when u did a zoom scale, and u'll figure it out. :)

**Comment by Ricardo1980** [comment]
Thanks, I did it, it seems it's 16 (mine is iPhone 5s). Does this number change from iPhone type to another? (is it the same in the iphone 4 than the iphone 6?)

Another question, sometimes it seems this code does not work (no zoom is done):

```
- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view {
            NSUInteger newZoom=mapView.zoomLevel+1;
            [mapView setCenterCoordinate:view.annotation.coordinate zoomLevel:newZoom animated:YES];
}
```

However this works (zoom is always done):

```
- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view {
            NSUInteger newZoom=mapView.zoomLevel+2;
            [mapView setCenterCoordinate:view.annotation.coordinate zoomLevel:newZoom animated:YES];
}
```

Maybe there is a floating point precision problem, do you have any idea?
Thanks a lot.

**Comment by Kjuly** [comment]
All devices should be same, I just tested on my iPhone 5 & iPad, works well, and same on simulators either. :)

And for ur 2nd question, maybe u can check whether the code below works (not tested):

``` Obj-C
- (void)mapView:(MKMapView *)mapView didSelectAnnotationView:(MKAnnotationView *)view {
  [mapView setCenterCoordinate:view.annotation.coordinate zoomLevel:mapView.zoomLevel animated:NO];
  NSUInteger newZoom = mapView.zoomLevel + 1;
  [mapView setCenterCoordinate:view.annotation.coordinate zoomLevel:newZoom animated:YES];
}
```

I guess before u set new zoom level for ur map, the zoom leave somehow changed when u press the annotation.

**Comment by Ricardo1980** [comment]
Nope, it does not work. It enters in something like a loop, very strange.

**Comment by Kjuly** [comment]
Can u offer a demo project for ur case, so i can help to debug it, or u can do more debugging by yourself, check the zoom level before & after ur tapping on annotation, and make sure ur `-mapView: didSelectAnnotationView:` only be invoked once.

**Comment by Ricardo1980** [comment]
Yes, on Monday I will do a demo. Thanks.
